plugin/hover.py

```diff
@@ -114,12 +119,30 @@ def run_async() -> None:
         sublime.set_timeout_async(run_async)
 
     def request_symbol_hover_async(self, listener: AbstractViewListener, point: int) -> None:
-        session = listener.session_async('hoverProvider', point)
-        if session:
-            document_position = text_document_position_params(self.view, point)
-            session.send_request_async(
-                Request("textDocument/hover", document_position, self.view),
-                lambda response: self.handle_response(listener, response, point))
+        sessions = list(listener.sessions_async('hoverProvider'))
+        hover_promises = []  # type: List[Promise[ResolvedHover]]
+        document_position = text_document_position_params(self.view, point)
+        for session in sessions:
+            hover_promises.append(session.send_request_task(
+                Request("textDocument/hover", document_position, self.view)
+            ))
+
+        Promise.all(hover_promises).then(lambda responses: self._on_all_settled(responses, listener, point))
```

Comment:
I'm not sure how exactly this "Promise" line works and haven't tested it, but I assume that will show the hover popup only after the last of the responses has arrived. The `_show_hover` method already seems to have some logic to update the popup if already visible, so would it maybe be better to already show the popup after the first response and then just update the popup content when new responses arrive? This would mean to for example store the hover content in an instance variable, update it with each new response and clear the variable after the last response arrived. That would decrease the delay if a certain server takes a longer time than other servers to answer. (But not sure whether it would be worth the additional logic, and how many users use multiple servers on a single language in general.)

Reply:
On the other side, it would provide worse experience because content would be changing and jumping around after the hover popup is shown. More than it already can at least.

Reply:
For now I would not do the update of the popup as soon as one response is received
Only if someone reports this as an issue, then I would see if I need to change the logic.

Reply:
Okay, sounds good to me.